website/src/components/FileEmbed/fileEmbed.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react'
+import SFGovIcon from '../SFGovIcon'
+import FigmaLogo from './img/figma_logo'
+import StorybookLogo from './img/storybook_logo'
+import IconExternalLink from '@theme/Icon/ExternalLink'
+
+// https://design-system.sf.gov/storybook/iframe.html?args=&id=components-buttons--primary-button&viewMode=story
+
+// https://www.figma.com/file/nCDNClTAztpLol9l74QWSP/Design-System-Components?node-id=3647%3A1504&t=Rvw9KsvFxlstnv3g-4
+
+const FileEmbed = ({ embedURL }) => {
+  const [hoverState, updateHoverState] = useState(false)
+  const [linkType, updateLinkType] = useState(false)
+
+  const figmaRegex = /https:\/\/([\w\.-]+\.)?figma.com\/(file|proto)\/([0-9a-zA-Z]{22,128})(?:\/.*)?$/
+
+  const storybookRegex = /story/i
+
+  const checkLink = (embedURL) => {
+    if (figmaRegex.test(embedURL)) {
+      updateLinkType('figma')
+    } else if (storybookRegex.test(embedURL)) {
+      updateLinkType('storybook')
+    } else {
+      updateLinkType(null)
+    }
+  }
+
+  useEffect(() => {
+    checkLink(embedURL)
+  }, [])
```

Comment:
since `linkType` is based off of just a single arg, you could set the value without state or effects, e.g.

```suggestion
  let linkType = null

  const figmaRegex = /https:\/\/([\w\.-]+\.)?figma.com\/(file|proto)\/([0-9a-zA-Z]{22,128})(?:\/.*)?$/

  const storybookRegex = /story/i

  if (figmaRegex.test(embedURL)) {
    linkType = 'figma'
  } else if (storybookRegex.test(embedURL)) {
    linkType = 'storybook'
  }
```